### DISCUSSION TITLE
API Integration

### DISCUSSION THREAD

**Discussion by lcunha1998** [PrivateBin/PrivateBin#1377] [Q&A] [answered]
### Describe the problem/question

I  need to integrate api via powershell and i am without success getting the response url

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

1.# Variables
$privateBinUrl = "https://privatebin.shared.cloud.int/api/v2/paste"
$content = "This is the content of your paste"
$pasteData = @{
    v = 2
    a = 0
    paste = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes($content))
    expire = "1day"  # options: "5min", "10min", "1hour", "1day", "1week", "1month", "1year", "never"
}
# Convert data to JSON
$jsonData = $pasteData | ConvertTo-Json -Compress

# Ignore SSL errors (only for testing purposes, not recommended for production)
Add-Type @"
using System.Net;
using System.Security.Cryptography.X509Certificates;

public class TrustAllCertsPolicy : ICertificatePolicy {
    public bool CheckValidationResult(
        ServicePoint srvPoint, X509Certificate certificate,
        WebRequest request, int certificateProblem) {
        return true;
    }
}
"@
[System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertsPolicy

try {
    # Send POST request
    $response = Invoke-RestMethod -Uri $privateBinUrl -Method Post -Body $jsonData -ContentType "application/json"
    # Output response
    Write-Output "Paste created successfully: $($response.url)"
} catch {
    Write-Error "Failed to create paste: $_"
}

2.
Paste created successfully: 
but doesnt show the paste

### What happens

no link is generated

### What should happen

give me a link

### Server address

https://privatebin.shared.cloud.int/

### PrivateBin version

1.5.1

### Issue reproducibility

No, I cannot reproduce it on https://privatebin.net.

**Comment by elrido** [comment, marked as answer]
Two things:

1. You should get back a JSON with a partial URL, which you then combine with the base58-endoced key you used to encrypt the message, to get the full URL:

From: https://github.com/PrivateBin/PrivateBin/wiki/API
```json
{"status": 0, "id": "[pasteID]", "url": "[serverAddress?pasteID]", "deletetoken": "[deleteToken]"}
```

More details on the URL format: https://github.com/PrivateBin/PrivateBin/wiki/Encryption-format#url-format-of-the-paste

2. Someone had shared their [PowerShell API implementation in an issue last year](https://github.com/PrivateBin/PrivateBin/issues/827#issuecomment-1434837076) - Are you using theirs or is this a new implementation?
